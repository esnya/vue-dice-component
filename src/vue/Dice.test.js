/* eslint-env jest */
/* eslint import/no-extraneous-dependencies: [error, { devDependencies: true }] */
/* eslint global-require: off */

describe('Dice', () => {
  function render(options) {
    const Vue = require('vue');
    const Dice = require('./Dice.vue');

    return new Vue({
      components: { Dice },
      render: createElement => createElement(Dice, options),
    }).$mount();
  }

  it('should render', () => {
    const vm = render();
    expect(vm.$el).toMatchSnapshot();
  });

  it('should render D6', () => {
    const vm = render({
      props: {
        faces: 6,
        angle: 2,
      },
    });
    expect(vm.$el).toMatchSnapshot();
  });

  it('should render D10', () => {
    const vm = render({
      props: {
        faces: 10,
        angle: 8,
      },
    });
    expect(vm.$el).toMatchSnapshot();
  });
});
