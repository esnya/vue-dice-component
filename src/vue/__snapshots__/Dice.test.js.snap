// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dice should render 1`] = `
<div
  class="dice d1"
>
  <svg
    class="dice-face"
  >
    <g
      class="dice-surface"
    >
      <!---->
    </g>
    <g
      class="dice-number"
    >
      <text
        x="30"
        y="30"
      >
        1
      </text>
    </g>
  </svg>
</div>
`;

exports[`Dice should render D6 1`] = `
<div
  class="dice d6 a2"
>
  <svg
    class="dice-face"
  >
    <g
      class="dice-surface"
    >
      <rect
        height="60"
        width="60"
      />
    </g>
    <g
      class="dice-number"
    >
      <circle
        cx="30"
        cy="30"
        r="10"
      />
    </g>
  </svg>
  <svg
    class="dice-face"
  >
    <g
      class="dice-surface"
    >
      <rect
        height="60"
        width="60"
      />
    </g>
    <g
      class="dice-number"
    >
      <circle
        cx="15"
        cy="15"
        r="6"
      />
      <circle
        cx="45"
        cy="45"
        r="6"
      />
    </g>
  </svg>
  <svg
    class="dice-face"
  >
    <g
      class="dice-surface"
    >
      <rect
        height="60"
        width="60"
      />
    </g>
    <g
      class="dice-number"
    >
      <circle
        cx="15"
        cy="15"
        r="6"
      />
      <circle
        cx="30"
        cy="30"
        r="6"
      />
      <circle
        cx="45"
        cy="45"
        r="6"
      />
    </g>
  </svg>
  <svg
    class="dice-face"
  >
    <g
      class="dice-surface"
    >
      <rect
        height="60"
        width="60"
      />
    </g>
    <g
      class="dice-number"
    >
      <circle
        cx="15"
        cy="15"
        r="6"
      />
      <circle
        cx="45"
        cy="45"
        r="6"
      />
      <circle
        cx="15"
        cy="45"
        r="6"
      />
      <circle
        cx="45"
        cy="15"
        r="6"
      />
    </g>
  </svg>
  <svg
    class="dice-face"
  >
    <g
      class="dice-surface"
    >
      <rect
        height="60"
        width="60"
      />
    </g>
    <g
      class="dice-number"
    >
      <circle
        cx="15"
        cy="15"
        r="6"
      />
      <circle
        cx="45"
        cy="45"
        r="6"
      />
      <circle
        cx="15"
        cy="45"
        r="6"
      />
      <circle
        cx="45"
        cy="15"
        r="6"
      />
      <circle
        cx="30"
        cy="30"
        r="6"
      />
    </g>
  </svg>
  <svg
    class="dice-face"
  >
    <g
      class="dice-surface"
    >
      <rect
        height="60"
        width="60"
      />
    </g>
    <g
      class="dice-number"
    >
      <circle
        cx="15"
        cy="15"
        r="6"
      />
      <circle
        cx="45"
        cy="45"
        r="6"
      />
      <circle
        cx="15"
        cy="45"
        r="6"
      />
      <circle
        cx="45"
        cy="15"
        r="6"
      />
      <circle
        cx="15"
        cy="30"
        r="6"
      />
      <circle
        cx="45"
        cy="30"
        r="6"
      />
    </g>
  </svg>
</div>
`;

exports[`Dice should render D10 1`] = `
<div
  class="dice d10 a8"
>
  <svg
    class="dice-face"
  >
    <g
      class="dice-surface"
    >
      <polygon
        points="-21.58019401015953 42.42640687119285 0 0 21.58019401015953 42.42640687119285"
      />
    </g>
    <g
      class="dice-number"
    >
      <text
        x="30"
        y="30"
      >
        1
      </text>
    </g>
  </svg>
  <svg
    class="dice-face"
  >
    <g
      class="dice-surface"
    >
      <polygon
        points="-21.58019401015953 42.42640687119285 0 0 21.58019401015953 42.42640687119285"
      />
    </g>
    <g
      class="dice-number"
    >
      <text
        x="30"
        y="30"
      >
        2
      </text>
    </g>
  </svg>
  <svg
    class="dice-face"
  >
    <g
      class="dice-surface"
    >
      <polygon
        points="-21.58019401015953 42.42640687119285 0 0 21.58019401015953 42.42640687119285"
      />
    </g>
    <g
      class="dice-number"
    >
      <text
        x="30"
        y="30"
      >
        3
      </text>
    </g>
  </svg>
  <svg
    class="dice-face"
  >
    <g
      class="dice-surface"
    >
      <polygon
        points="-21.58019401015953 42.42640687119285 0 0 21.58019401015953 42.42640687119285"
      />
    </g>
    <g
      class="dice-number"
    >
      <text
        x="30"
        y="30"
      >
        4
      </text>
    </g>
  </svg>
  <svg
    class="dice-face"
  >
    <g
      class="dice-surface"
    >
      <polygon
        points="-21.58019401015953 42.42640687119285 0 0 21.58019401015953 42.42640687119285"
      />
    </g>
    <g
      class="dice-number"
    >
      <text
        x="30"
        y="30"
      >
        5
      </text>
    </g>
  </svg>
  <svg
    class="dice-face"
  >
    <g
      class="dice-surface"
    >
      <polygon
        points="-21.58019401015953 42.42640687119285 0 0 21.58019401015953 42.42640687119285"
      />
    </g>
    <g
      class="dice-number"
    >
      <text
        x="30"
        y="30"
      >
        6
      </text>
    </g>
  </svg>
  <svg
    class="dice-face"
  >
    <g
      class="dice-surface"
    >
      <polygon
        points="-21.58019401015953 42.42640687119285 0 0 21.58019401015953 42.42640687119285"
      />
    </g>
    <g
      class="dice-number"
    >
      <text
        x="30"
        y="30"
      >
        7
      </text>
    </g>
  </svg>
  <svg
    class="dice-face"
  >
    <g
      class="dice-surface"
    >
      <polygon
        points="-21.58019401015953 42.42640687119285 0 0 21.58019401015953 42.42640687119285"
      />
    </g>
    <g
      class="dice-number"
    >
      <text
        x="30"
        y="30"
      >
        8
      </text>
    </g>
  </svg>
  <svg
    class="dice-face"
  >
    <g
      class="dice-surface"
    >
      <polygon
        points="-21.58019401015953 42.42640687119285 0 0 21.58019401015953 42.42640687119285"
      />
    </g>
    <g
      class="dice-number"
    >
      <text
        x="30"
        y="30"
      >
        9
      </text>
    </g>
  </svg>
  <svg
    class="dice-face"
  >
    <g
      class="dice-surface"
    >
      <polygon
        points="-21.58019401015953 42.42640687119285 0 0 21.58019401015953 42.42640687119285"
      />
    </g>
    <g
      class="dice-number"
    >
      <text
        x="30"
        y="30"
      >
        0
      </text>
    </g>
  </svg>
</div>
`;
